content/en/docs/project-user-guide/image-builder/s2i-templates.md

```diff
@@ -6,4 +6,323 @@ linkTitle: "Customize S2I Templates"
+
+   ```bash
+   $ make build
+   docker build -t kubespheredev/nginx-centos7-s2ibuilder-sample .
+   Sending build context to Docker daemon  164.9kB
+   Step 1/17 : FROM kubespheredev/s2i-base-centos7:1
+    ---> 48f8574c05df
+   Step 2/17 : LABEL maintainer="Runze Xia <runzexia@yunify.com>"
+    ---> Using cache
+    ---> d60ebf231518
+   Step 3/17 : ENV NGINX_VERSION=1.6.3
+    ---> Using cache
+    ---> 5bd34674d1eb
+   Step 4/17 : LABEL io.k8s.description="Nginx Webserver"       io.k8s.display-name="Nginx 1.6.3"       io.kubesphere.expose-services="8080:http"       io.kubesphere.tags="builder,nginx,html"
+    ---> Using cache
+    ---> c837ad649086
+   Step 5/17 : RUN yum install -y epel-release &&     yum install -y --setopt=tsflags=nodocs nginx &&     yum clean all
+    ---> Running in d2c8fe644415
+   
+   …………
+   …………
+   …………
+   
+   Step 17/17 : CMD ["/usr/libexec/s2i/usage"]
+    ---> Running in c24819f6be27
+   Removing intermediate container c24819f6be27
+    ---> c147c86f2cb8
+   Successfully built c147c86f2cb8
+   Successfully tagged kubespheredev/nginx-centos7-s2ibuilder-sample:latest
+   ```
+
+3. With the Image Builder created, run the following command to create an application image.
+
+   ```bash
+   $ s2i build ./test/test-app kubespheredev/nginx-centos7-s2ibuilder-sample:latest sample-app
+   ---> Building and installing application from source...
+   Build completed successfully
+   ```
+
+   {{< notice note >}}
+
+   Following the logic defined in the `assemble` script, S2I creates an application image using the Image Builder as a base and injecting the source code from the `test/test-app` directory.
+
+   {{</ notice >}}
+
+4. Run the following command to run the application image.
+
+   ```bash
+   docker run -p 8080:8080  sample-app
+   ```
+
+   You can access the Nginx application at `http://localhost:8080`.
+
+   ![access-nginx](/images/docs/project-user-guide/image-builder/s2i-templates/access-nginx.png)
+
+### Step 5: Push image and create S2I template
+
+Once you finish testing the S2I Image Builder locally, you can push the image to your custom image repository. You also need to create a YAML file as the S2I Builder template as follows.
+
+#### s2ibuildertemplate.yaml
+
+```yaml
+apiVersion: devops.kubesphere.io/v1alpha1
+kind: S2iBuilderTemplate
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+    builder-type.kubesphere.io/s2i: "s2i"
+  name: nginx-demo
+spec:
+  containerInfo:
+    - builderImage: kubespheredev/nginx-centos7-s2ibuilder-sample
+  codeFramework: nginx # type of code framework
+  defaultBaseImage: kubespheredev/nginx-centos7-s2ibuilder-sample # default Image Builder (can be replaced by customized image)
+  version: 0.0.1 # Builder template version
+  description: "This is a S2I builder template for Nginx builds whose result can be run directly without any further application server.." # Builder template description
```

Comment:
This is a S2I builder template for Nginx builds whose result can be run directly without any further application server..
->
This is an S2I builder template for NGINX builds whose result can be run directly without any further application server.